web/app/habit/stake/[challengeId]/page.tsx

```diff
@@ -1,325 +1,13 @@
-            </div>
-          </>
-        )}
-
-        {/**
-         * Disable button when challenge hasn't selected or when not enough balance
-         * If support batch tx -> Join with Batch Tx (Approve -> Join)
-         * If doesn't support batch tx, has enough balance, has enough allowance -> Stake Tx
-         * If doesn't support batch tx, has enough balance, not enough allowance -> Approve Tx
-         */}
-        {challenge && (
-          <button
-            type="button"
-            className="bg-primary bg-primary mt-4 rounded-lg border-solid px-6 py-3 font-bold disabled:cursor-not-allowed disabled:bg-gray-400"
-            style={{ width: '250px', height: '45px', color: 'white' }}
-            onClick={
-              currentChainSupportBatchTx || hasEnoughAllowance
-                ? onJoinButtonClick
-                : onApproveTestTokenClick
-            }
-            disabled={
-              !challenge ||
-              !hasEnoughBalance ||
-              mintPending ||
-              approvePending ||
-              joinPending ||
-              joinPendingInBatchTx ||
-              isJoinLoading ||
-              isMintLoading ||
-              isApproveLoading
-            }
-          >
-            {/**
-             * Display only when challenge is selected
-             * If doesn't have enough balance -> Display Approve
-             * If has enough allowance -> Display Stake
-             */}
-            {hasEnoughAllowance || currentChainSupportBatchTx ? `Join` : 'Approve'}
-          </button>
-        )}
-
-        {/**
-         * Display only when challenge is selected
-         * If doesn't have enough balance -> Mint first
-         * If has enough balance -> Show balance
-         */}
-        <div className="p-4 text-xs">
-          {!challenge || loading ? (
-            <p> Loading Challenge </p>
-          ) : testTokenBalance && !hasEnoughBalance ? (
-            <p>
-              {' '}
-              🚨 Insufficient Balance: {testTokenBalance
-                ? formatEther(testTokenBalance.value)
-                : 0}{' '}
-              ALI.{' '}
-              <span className="font-bold hover:underline" onClick={onMintTestTokenClick}>
-                {' '}
-                Mint Test Token now{' '}
-              </span>{' '}
-            </p>
-          ) : (
-            <p>
-              {' '}
-              💰 Smart Wallet Balance: {testTokenBalance
-                ? formatEther(testTokenBalance.value)
-                : 0}{' '}
-              ALI.{' '}
-            </p>
-          )}
-        </div>
-      </div>
-    </main>
-  );
+import { generateMetadata } from '@/utils/generateMetadata';
+import StakeContent from './stakeContent';
+
+export const metadata = generateMetadata({
+  title: 'Join',
+  description: 'Habit Builder for Happy Builders',
+  images: 'themes.png',
+  pathname: '',
+});
+
+export default function ChallengeListPage() {
```

Comment:
moved everything in `page` to `StakeContent` because we need to declare metadata in the `page.tsx` file and it cannot has "use client"